src_lockfree/michael_scott_queue.ml

```diff
@@ -45,6 +47,27 @@ let pop { head; _ } =
   in
   loop ()
 
+let pop { head; _ } =
+  let b = Backoff.create () in
+  let rec loop () =
+    let old_head = Atomic.get head in
+    match Atomic.get old_head with
+    | Nil -> raise Empty
+    | Next (value, next) when Atomic.compare_and_set head old_head next -> value
+    | _ ->
+        Backoff.once b;
+        loop ()
+  in
+  loop ()
```

Comment:
This is the same function as `pop_opt` except for very small changes. Should I use `[@inline]` and a generic pop function ?